File: typescript/iniciante/1015.ts
import { bold, cyan } from "./../fmt/colors";

function readCoordinates(promptMessage: string): [number, number] {
  const components = (prompt(promptMessage) ?? "").split(" ");
  const x = Number(components[0]?.replace(",", "."));
  const y = Number(components[1]?.replace(",", "."));
  return [x, y];
}

function calculateDistance(
  x1: number,
  y1: number,
  x2: number,
  y2: number,
): number {
  return Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
}

const [X1, Y1] = readCoordinates("Digite X1 e Y1 separados por espaço:");
const [X2, Y2] = readCoordinates("Digite X2 e Y2 separados por espaço:");

const distance = calculateDistance(X1, Y1, X2, Y2);

console.log(cyan(bold(distance.toFixed(4))));

File: swift/iniciante/1015.swift
import Foundation

func readCoordinates() -> (Double, Double)? {
    guard let input = readLine() else { return nil }
    let components = input.split(separator: " ").map { String($0) }
    guard components.count == 2,
          let x = Double(components[0].replacingOccurrences(of: ",", with: ".")),
          let y = Double(components[1].replacingOccurrences(of: ",", with: "."))
    else {
        return nil
    }
    return (x, y)
}

func calculateDistance(x1: Double, y1: Double, x2: Double, y2: Double) -> Double {
    return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2))
}

guard let (X1, Y1) = readCoordinates() else {
    print("Erro na entrada das primeiras coordenadas.")
    exit(1)
}

guard let (X2, Y2) = readCoordinates() else {
    print("Erro na entrada das segundas coordenadas.")
    exit(1)
}

let distance = calculateDistance(x1: X1, y1: Y1, x2: X2, y2: Y2)

print(String(format: "%.4f", distance))
